test/unit/org/apache/cassandra/cql3/validation/entities/TupleTypeTest.java

```diff
@@ -233,5 +258,136 @@ public void testReversedTypeTuple() throws Throwable
         DateFormat df = new SimpleDateFormat("yyyy-MM-dd HH:mmX", Locale.ENGLISH);
         assertRows(execute("SELECT tdemo FROM %s"), row(tuple( df.parse("2017-02-03 03:05+0000"), "Europe")));
     }
+
+    @Test
+    public void tuplePkReadWrite()
+    {
+        qt().withExamples(1).withShrinkCycles(0).forAll(caseGen()).checkAssert(orFail(testcase -> {
```

Comment:
A couple questions here:

1.) Running this test with multiple examples seems to produce the same progression of `TupleType` instances. Is there a seed buried somewhere?

2.) Most of the time, a randomized test is valuable because it can examine a fairly large space of possible inputs in a single run. (i.e. It should be likely that a single run would surface most problems, even if it's not guaranteed to do so, since it's...random.) As far as I can tell here, with one example, we'd have to run `TupleTypeTest` itself many times to dig up anything interesting.

Reply:
1) qt has a concept of "shrink target" which appears to be the first test it tries, so believe this is why you see the same thing each time

2) agree.  I increased the numbers now and tried to get the tests to run no more than 10s.  I had 10 before but saw timeouts in CI, I didn't put 2 and 2 together to relieve that the timeouts were caused by the unbounded retries for distinct rows when the types didn't have 10 distinct rows in the domain (tuple<boolean> can have at most 2 rows)